src/main/java/com/google/sps/servlets/ProfileServlet.java

```diff
@@ -0,0 +1,121 @@
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.sps.data.Student;
+import com.google.sps.data.TimeRange;
+import com.google.sps.data.Tutor;
+import com.google.sps.data.TutorSession;
+import com.google.sps.utilities.TutorSessionDatastoreService;
+import com.google.sps.utilities.RealTutorSessionDatastore;
+import com.google.sps.utilities.MockTutorSessionDatastore;
+import com.google.sps.utilities.AvailabilityDatastoreService;
+import com.google.sps.utilities.MockAvailabilityDatastore;
+import com.google.sps.utilities.RealAvailabilityDatastore;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Optional;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles user's registration info */
+@WebServlet("/profile")
+public class ProfileServlet extends HttpServlet {
+
+    private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private TutorSessionDatastoreService sessionDatastore;
+    private AvailabilityDatastoreService availabilityDatastore;
+
+    /**
+    * Because we created a constructor with a parameter (the testing one), the default empty constructor does not work anymore so we have to explicitly create it. 
+    * We need the default one for deployment because the servlet is created without parameters.
+    */
+    public ProfileServlet(){}
+
+    public ProfileServlet(boolean test) {
+        if(test) {
+            sessionDatastore = new MockTutorSessionDatastore();
+            availabilityDatastore = new MockAvailabilityDatastore();
+        }
+    }
+
+    public void init() {
+        sessionDatastore = new RealTutorSessionDatastore();
+        availabilityDatastore = new RealAvailabilityDatastore();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json");
+        
+        String userId = Optional.ofNullable(request.getParameter("user-id")).orElse(null);
+        
+        // Find out whether the user is a student or a tutor
+        Query query = new Query("User").setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
+        PreparedQuery results = datastore.prepare(query);
+        Entity entity = results.asSingleEntity();
+        String role = (String) entity.getProperty("role"); 
+
+        // User is a student, get their info
+        if (role.toLowerCase().equals("student")) {
```

Comment:
We definitely don't have to do this now, but it might be easier if we had Student and Tutor as a subclass of User since they have some of the same basic properties for profiles. What do you think?

Reply:
Yes that would be easier. Then we wouldn't have to repeat code for things like name and email.